src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -399,21 +399,35 @@ private ISet<DocumentId> GetProjectOpenDocuments_NoLock(ProjectId project)
             DocumentId documentId, SourceTextContainer textContainer,
             bool isCurrentContext = true)
         {
-            OnDocumentOpenedAsync(documentId, textContainer, isCurrentContext).Wait(CancellationToken.None);
+            OnDocumentOpenedAsync(documentId, textContainer, isCurrentContext, wait: true).Wait(CancellationToken.None);
         }
 
-        internal async Task OnDocumentOpenedAsync(
-            DocumentId documentId, SourceTextContainer textContainer, 
+        internal Task OnDocumentOpenedAsync(
+            DocumentId documentId, SourceTextContainer textContainer,
             bool isCurrentContext = true)
         {
+            return OnDocumentOpenedAsync(documentId, textContainer, isCurrentContext, wait: false);
+        }
+
+        private async Task OnDocumentOpenedAsync(
+            DocumentId documentId, SourceTextContainer textContainer, 
+            bool isCurrentContext,
+            bool wait)
```

Comment:
In the 'async' case.  In that case we want to 'await' instead.  This prevents the problem of having sync code call the async version and then immediately block on it.